test/spec.js

```diff
@@ -1,132 +1,167 @@
-global.window = global // mock
+global.window = global; // mock
 
-const test = require('tape')
-const sinon = require('sinon')
+const sinon = require('sinon');
+const test = require('tape');
 
-const detectProvider = require('../dist')
+const detectProvider = require('../dist');
 
 // test mocking utility
 const mockGlobalProps = (ethereum) => {
-  window.addEventListener = sinon.fake()
-  window.removeEventListener = sinon.fake()
-  console.error = sinon.fake()
+  global.addEventListener = sinon.fake();
```

Comment:
ESLint was complaining that `window` was unavailable in Node tests. We do set it at the top of this file, but since it's `global`, I opted to change all instances of `window` to `global` in this file.